### DISCUSSION TITLE
[nuxt] [request error] [unhandled] [500] Named export 'placements' not found. The requested module '@popperjs/core' is a CommonJS module, which may not support all module.exports as named exports.

### DISCUSSION THREAD

**Discussion by u007** [nuxt/nuxt#10822] [Questions (Nuxt 2)]
hi,

im still struggling trying to understand how to work with nuxt.
i tried to add alias to nuxt config to point this to the esm version,

```
alias: {
    '@popperjs/core': 'node_modules/@popperjs/core/dist/esm/index.js'
  },
```

but now I'm getting same error with
```
pnpm run build
pnpm run preview
```

here is my package.json
```
{
  "private": true,
  "type": "module",
  "scripts": {
    "build": "nuxt build",
    "dev": "nuxt dev -p 3800",
    "generate": "nuxt generate",
    "preview": "nuxt preview",
    "postinstall": "nuxt prepare && npx prisma generate",
    "lint": "eslint --ext .ts,.js,.vue . --fix"
  },
  "devDependencies": {
    "@iconify-json/ep": "^1.1.8",
    "@nuxtjs/eslint-config-typescript": "^12.0.0",
    "@types/bcrypt": "^5.0.0",
    "@types/jsonwebtoken": "^8.5.9",
    "@types/qs": "^6.9.7",
    "eslint": "^8.28.0",
    "eslint-plugin-nuxt": "^4.0.0",
    "nuxt": "3.0.0",
    "nuxt-windicss": "^2.5.5",
    "prisma": "^4.6.1",
    "sass": "^1.56.1",
    "typescript": "^4.9.3",
    "unplugin-auto-import": "^0.11.4",
    "unplugin-icons": "^0.14.13",
    "unplugin-vue-components": "^0.22.9",
    "vite-plugin-inspect": "^0.7.8",
    "vue-tsc": "^1.0.9"
  },
  "dependencies": {
    "@ant-design/icons-vue": "^6.1.0",
    "@element-plus/icons-vue": "^2.0.10",
    "@pinia/nuxt": "^0.4.4",
    "@popperjs/core": "^2.11.6",
    "@prisma/client": "^4.6.1",
    "@trpc/client": "10.0.0-rc.8",
    "@trpc/server": "10.0.0-rc.8",
    "@vee-validate/zod": "^4.7.3",
    "ant-design-vue": "^3.2.15",
    "axios": "^1.1.3",
    "bcrypt": "^5.1.0",
    "bson": "^4.7.0",
    "element-plus": "^2.2.22",
    "h3": "^0.8.6",
    "h3-zod": "^0.3.10",
    "jsonwebtoken": "^8.5.1",
    "pinia-plugin-persistedstate": "^2.4.0",
    "pnpm": "^7.16.1",
    "qs": "^6.11.0",
    "superjson": "^1.11.0",
    "trpc-nuxt": "0.4.2",
    "vee-validate": "^4.7.3",
    "vue-accessible-color-picker": "^4.1.2",
    "zod": "3.19.1"
  }
}
```

error:
```
[nuxt] [request error] [unhandled] [500] Named export 'placements' not found. The requested module '@popperjs/core' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from '@popperjs/core';
const { placements } = pkg;

  at ModuleJob._instantiate (node:internal/modules/esm/module_job:128:21)  
  at async ModuleJob.run (node:internal/modules/esm/module_job:194:5)  
  at async Promise.all (index 0)  
  at async ESMLoader.import (node:internal/modules/esm/loader:385:24)  
  at async ./server/chunks/handlers/renderer.mjs:309:24  
  at async ./server/chunks/handlers/renderer.mjs:376:64  
  at async ./server/chunks/handlers/renderer.mjs:28:22  
  at async Object.handler (./server/chunks/nitro/node-server.mjs:658:19)  
  at async Server.toNodeHandle (./server/chunks/nitro/node-server.mjs:708:7)
```

**Comment by ryiot** [comment]
Last login: Wed Dec 14 03:08:42 on ttys008
➜  test node ./server/index.mjs
Listening http://[::]:3000
[nuxt] [request error] [unhandled] [500] Cannot find package '@popperjs/core' imported from /Users/jack/Desktop/test/server/node_modules/element-plus/es/components/popper/src/content.mjs
  at new NodeError (node:internal/errors:371:5)  
  at packageResolve (node:internal/modules/esm/resolve:932:9)  
  at moduleResolve (node:internal/modules/esm/resolve:978:18)  
  at defaultResolve (node:internal/modules/esm/resolve:1080:11)  
  at ESMLoader.resolve (node:internal/modules/esm/loader:530:30)  
  at ESMLoader.getModuleJob (node:internal/modules/esm/loader:251:18)  
  at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:79:40)  
  at link (node:internal/modules/esm/module_job:78:36)
[nuxt] [request error] [unhandled] [500] Cannot find package '@popperjs/core' imported from /Users/jack/Desktop/test/server/node_modules/element-plus/es/components/popper/src/content.mjs
  at new NodeError (node:internal/errors:371:5)  
  at packageResolve (node:internal/modules/esm/resolve:932:9)  
  at moduleResolve (node:internal/modules/esm/resolve:978:18)  
  at defaultResolve (node:internal/modules/esm/resolve:1080:11)  
  at ESMLoader.resolve (node:internal/modules/esm/loader:530:30)  
  at ESMLoader.getModuleJob (node:internal/modules/esm/loader:251:18)  
  at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:79:40)  
  at link (node:internal/modules/esm/module_job:78:36)

**Comment by ryiot** [comment]
<img width="570" alt="image" src="https://user-images.githubusercontent.com/78907118/207424891-63ea1902-44ca-4a83-ba60-bb2e1509a305.png">

**Comment by ryiot** [comment]
<img width="1337" alt="image" src="https://user-images.githubusercontent.com/78907118/207425155-8d28e99c-d9ea-4f61-a2e6-bae783859f7e.png">
![image](https://user-images.githubusercontent.com/78907118/207425228-2b92d44c-9a99-4b7f-8a11-b78e4285ade0.png)

**Comment by ryiot** [comment]
![image](https://user-images.githubusercontent.com/78907118/207425296-704af18d-c2bb-45e8-8986-9aabe75736b5.png)

**Comment by ryiot** [comment]
<img width="570" alt="image" src="https://user-images.githubusercontent.com/78907118/207425613-8d1e7ae6-8ca6-4952-b0c9-23c50345a8a7.png">

**Comment by ryiot** [comment]
<img width="669" alt="image" src="https://user-images.githubusercontent.com/78907118/207425789-ef4015f3-13db-4a66-8812-6b897a01e7a5.png">

**Comment by ryiot** [comment]
https://github.com/element-plus/element-plus/issues?q=is%3Aissue+is%3Aopen+nuxt3

**Comment by ryiot** [comment]
https://github.com/element-plus/element-plus/issues/10979

**Comment by szulcus** [comment]
Now, I have the same problem with `humps` (used in nuxt layer). So far, aliases have solved the problem (https://github.com/nuxt/nuxt/issues/19426)
